unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/WearableGridItemComponentView.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UIComponents.Scripts.Components;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace DCL.Backpack
+{
+    public class WearableGridItemComponentView : BaseComponentView<WearableGridItemModel>
+    {
+        [Serializable]
+        internal struct RareBackground
+        {
+            public NftRarity rarity;
+            public GameObject container;
+        }
+
+        [SerializeField] internal GameObject selectedContainer;
+        [SerializeField] internal GameObject equippedContainer;
+        [SerializeField] internal GameObject hoverUnequippedContainer;
+        [SerializeField] internal GameObject hoverEquippedContainer;
+        [SerializeField] internal GameObject hoverSelectedUnequippedContainer;
+        [SerializeField] internal GameObject hoverSelectedEquippedContainer;
+        [SerializeField] internal GameObject isNewContainer;
+        [SerializeField] internal ImageComponentView image;
+        [SerializeField] internal Button interactButton;
+        [SerializeField] internal RareBackground[] backgroundsByRarityConfiguration;
+
+        private Dictionary<NftRarity, GameObject> backgroundsByRarity;
+        private string lastThumbnailUrl;
+
+        public event Action<WearableGridItemModel> OnSelected;
+        public event Action<WearableGridItemModel> OnEquipped;
+        public event Action<WearableGridItemModel> OnUnequipped;
+
+        public override void Awake()
+        {
+            base.Awake();
+
+            backgroundsByRarity = backgroundsByRarityConfiguration
+               .ToDictionary(background => background.rarity, background => background.container);
+
+            interactButton.onClick.AddListener(() =>
```

Comment:
for safety as we do everywhere I'd add before this a `interactButton.onClick.RemoveAllListeners();`

Reply:
since its in the `Awake` function i dont see the need because no subscriptions (via code) could exist before that point

Reply:
alrighty :)